Remove update_wrapper from reify

luhn opened the pull request:
Reify uses `update_wrapper`, which makes the descriptor look like the wrapped method.  However, this isn't the desired behavior, because reify changes the method to act like the property.

This causes problems with Sphinx autodoc as reported in #3655, because Sphinx sees the reify'd attribute as a method and throws an error when attempting to get a signature.  You can see the problem here:

```python
import inspect
from pyramid.decorator import reify
from functools import cached_property
from sphinx.util.inspect import isattributedescriptor


class Widget:
    @property
    def prop(self):
        return 'Foo'

    @cached_property
    def cached(self):
        return 'Foo'

    @reify
    def reify(self):
        return 'Foo'


print(isattributedescriptor(Widget.prop))
print(isattributedescriptor(Widget.cached))
print(isattributedescriptor(Widget.reify))
```

Output is:

```
True
True
False
```

This fix removes `update_wrapper` from reify and just sets `__doc__` manually.  This is in line with stdlib's implementation of [functools.cached_property](https://github.com/python/cpython/blob/ebe20d9e7eb138c053958bc0a3058d34c6e1a679/Lib/functools.py#L933).

Fixes #3655.

luhn commented:
So this causes [this test](https://github.com/Pylons/pyramid/blob/d76903c58c5d0dfd8edc9de40b8ad2382d4f5634/tests/test_config/test_views.py#L4298) to fail.

I'm confused what the test is doing.  Why are we reify'ing the `viewdefaults` wrapper?  That seems like a mistake.

mmerickel commented:
Huh, yeah so I traced that test back to https://github.com/Pylons/pyramid/commit/52fde9aa164979a5c9e547826c811e9b7116ae93. It looks to me like I totally screwed up the test case when porting it to the new location in the codebase.

luhn commented:
Since it looks like the original intent was to test reify, I've removed the failing test and added some new tests for reify.

However, I don't see any tests dedicated to viewdefaults...  Am I overlooking them or do we need to write some?

luhn commented:
Should this be backported?  Technically this isn't a bw compat change, but the differences are subtle.

mmerickel commented:
Not sure why they aren't showing up nicely in github search but they are here: https://github.com/Pylons/pyramid/blob/ee7ca28cc51cf40d1190144834704e287c9fc72d/tests/test_view.py#L917

mmerickel commented:
I think it's ok to backport if you want to.
